sources/web/datalab/polymer/modules/file-manager/file-manager.ts

```diff
@@ -151,71 +151,76 @@ interface FileManager {
    *               useful for downloading notebooks, which are by default read
    *               as JSON, which doesn't preserve formatting.
    */
-  getStringContent(fileId: DatalabFileId, asText?: boolean): Promise<string>;
+  abstract getStringContent(fileId: DatalabFileId, asText?: boolean): Promise<string>;
 
   /**
    * Returns a DatalabFile object for the root directory.
    */
-  getRootFile(): Promise<DatalabFile>;
+  abstract getRootFile(): Promise<DatalabFile>;
 
   /**
    * Saves the given string as a file's content.
    * @param file object containing information about the destination file to
    *             save to.
    * @param content string to be saved in the file
    */
-  saveText(file: DatalabFile, content: string): Promise<DatalabFile>;
+  abstract saveText(file: DatalabFile, content: string): Promise<DatalabFile>;
```

Comment:
Maybe most of these functions should have a default implementation that throws UnsupportedMethod.